validate.swift

```diff
@@ -125,6 +125,32 @@ extension Result where Success == Void {
 
 // MARK: Functions
 
+/**
+ Fetch the default branch for a given GitHub repository.
+ - Parameter userName: Repository owner
+ - Parameter repositoryName: Repository name
+ - Returns: default branch name if successful; otherwise `noResult` or a `decodingError`.
+ */
+func getGitHubDefaultBranch(for userName: String, repositoryName: String) -> Result<String, PackageError> {
+  struct Repository: Decodable {
+    var default_branch: String  // don't bother with re-casing this for now...
+  }
+  let sema = DispatchSemaphore(value: 0)
+  let apiURL = URL(string: "https://api.github.com/repos/\(userName)/\(repositoryName)")!
+  print("apiURL: \(apiURL.absoluteString)")
+  var result: Result<String, PackageError> = .failure(.noResult)
+  let task = URLSession.shared.dataTask(with: apiURL) { (data, response, error) in
+    guard let data = data else { return }
+    result = Result { try JSONDecoder().decode(Repository.self, from: data) }
+      .mapError { .decodingError($0) }
+      .map { $0.default_branch }
+    sema.signal()
+  }
+  task.resume()
+  let _ = sema.wait(timeout: DispatchTime.now() + .seconds(2))
+  return result
+}
```

Comment:
Not ideal to make this synchronous but it was the easier way to weave it in for now. This could be a whole asynchronous chain but that would have touched a lot more code... I'm hoping it's ok like this since the individual checks are properly asynchronous.